### PR TITLE
Refactor Kandinsky endpoint into diffusers

### PR DESCRIPTION
**Pull request by kodxana**
Changes:
- Switched Kandinsky endpoint into Diffusers pipeline
- Added xformers for memory optimization
- Added seed parameter default is -1 (random)
- Refactored model_fetcher.py to include Diffusers base model

Overlay Diffusers brings much faster image generation compared to orginal implementation.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,6 +23,7 @@ COPY builder/requirements.txt /requirements.txt
 RUN pip install --upgrade pip && \
     pip install -r /requirements.txt && \
     rm /requirements.txt
+RUN pip install diffusers transformers accelerate
```
Comment (justinmerrell) [inline Dockerfile:26]: Any reason these were not part of the requirements.txt?

**Review by justinmerrell — CHANGES_REQUESTED** [review]
Looks good overall, just curious about the requirements.txt

File: src/rp_handler.py [hunk 2]
```diff
@@ -1,66 +1,58 @@
...
+                     width=validated_input['w'],
+                     num_inference_steps=validated_input['num_steps'],
+                     guidance_scale=validated_input['guidance_scale']).images[0]
 
     # Save the generated image to a file
     os.makedirs(f"/{job['id']}", exist_ok=True)
 
     image_path = os.path.join(f"/{job['id']}", "0.png")
-    images[0].save(image_path)
+    image.save(image_path)
```
[the remark this answers is not in the document]
Comment (kodxana) [reply src/rp_handler.py]: Added:
- num_images to schema (also removed old batch_size)

File: builder/model_fetcher.py [hunk 3]
```diff
@@ -1,21 +1,16 @@
...
 
-cache_dir = os.path.expanduser("~/.cache/kandinsky2")
-clip_cache_dir = os.path.expanduser("~/.cache/clip")
+import torch
+from diffusers import DiffusionPipeline
 
-# Make directory if it doesn't exist
-os.makedirs(clip_cache_dir, exist_ok=True)
+def get_kandinsky_pipelines():
+    pipe_prior = DiffusionPipeline.from_pretrained("kandinsky-community/kandinsky-2-1-prior",
```
[the remark this answers is not in the document]
Comment (kodxana) [reply builder/model_fetcher.py:7]: Is same model just in Diffusion format

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (kodxana) [reply Dockerfile]: Moved diffusers transformers accelerate to requirements.txt with locked minimum version of diffusers

File: builder/requirements.txt
Comment (justinmerrell) [inline builder/requirements.txt:1]: Need to version lock:

xformers
transformers
accelerate

**Review by justinmerrell — CHANGES_REQUESTED** [review]
